sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py

```diff
@@ -134,14 +142,18 @@ async def __aenter__(self):
     async def __aexit__(self, *args):
         return await self.client_connection.pipeline_client.__aexit__(*args)
 
-    async def close(self):
-        # type: () -> None
+    async def close(self) -> None:
         """Close this instance of CosmosClient."""
         await self.__aexit__()
 
     @classmethod
-    def from_connection_string(cls, conn_str, credential=None, consistency_level=None, **kwargs):
-        # type: (str, Optional[Union[str, Dict[str, str]]], Optional[str], Any) -> CosmosClient
+    def from_connection_string(
+            cls: ClassType,
```

Comment:
I would think this should be `CosmosClient` since that is logically the only type that could be returned here.

Reply:
Your docstring :rtype: shows CosmosClient, so this should match.

Reply:
The only way to get this to work for me has been to import CosmosClient from it's same file, which seems redundant  - is that the way to do it? I haven't been able to resolve the definition otherwise

Reply:
It is unable to find the reference to CosmosClient when it is just put like this, so I went with ClassType the way I saw other SDKs do

Reply:
You can use a forward reference for `"CosmosClient"` if you need to. It would seem to me that if other SDKs are just using `ClassType` then they are incorrect.